spec/facility_spec.rb

```diff
@@ -2,25 +2,220 @@
+
+      expect(@bolt.registration_date).to be_a Date
+      expect(@bolt.plate_type).to be :ev
+      expect(@facility_1.collected_fees).to be 200
+    end
+
+    it "can accumulate registered vehicles and collected fees" do
+      @facility_1.add_service("Vehicle Registration")
+      @facility_1.register_vehicle(@cruz)
+      @facility_1.register_vehicle(@camaro)
+      @facility_1.register_vehicle(@bolt)
+
+      expect(@facility_1.registered_vehicles).to eq [@cruz, @camaro, @bolt]
+      expect(@facility_1.collected_fees).to be 325
+    end
+
+    it "will not register vehicle if service is not available at facility" do
+      expect(@facility_2.registered_vehicles).to eq []
+      expect(@facility_2.services).to eq []
+
+      @facility_2.register_vehicle(@bolt)
+      
+      expect(@facility_2.registered_vehicles).to eq []
+      expect(@facility_2.collected_fees).to be 0
+    end
+  end
+
+  describe "#administer_written_test" do
+    it "can will not administer test if facility does not offer service" do
+      @facility_1.administer_written_test(@registrant_1)
+      expect(@registrant_1.license_data).to eq ({written: false, license: false, renewed: false})
+    end
+      
+    it "can administer to registrant with permit and age 16 or older" do
+      expect(@registrant_1.license_data).to eq ({written: false, license: false, renewed: false})
+      expect(@registrant_1.permit?).to eq true
+
+      @facility_1.add_service("Written Test")
+      @facility_1.administer_written_test(@registrant_1)
+      
+      expect(@registrant_1.license_data).to eq ({written: true, license: false, renewed: false})
+    end
+
+    it "will not administer to registrant age 16 or older without a permit " do
+      @facility_1.add_service("Written Test")
+      
+      expect(@registrant_2.age).to be 16
+      expect(@registrant_2.permit?).to be false
+
+      @facility_1.administer_written_test(@registrant_2)
+
+      expect(@registrant_2.license_data).to eq ({written: false, license: false, renewed: false})
+    end
+
+    it "can administer to registrant age 16 or older who has earned a permit" do
+      @facility_1.add_service("Written Test")      
+      @registrant_2.earn_permit      
+      @facility_1.administer_written_test(@registrant_2)
+      
+      expect(@registrant_2.license_data).to eq ({written: true, license: false, renewed: false})
+    end
+
+    it "will not administer to registrant age 15 or younger" do
+      @facility_1.add_service("Written Test")   
+
+      expect(@registrant_3.age).to be 15
+      expect(@registrant_3.permit?).to be false
+
+      @facility_1.administer_written_test(@registrant_3)
+      
+      expect(@registrant_3.license_data).to eq ({written: false, license: false, renewed: false})
+
+      @registrant_3.earn_permit
+      @facility_1.administer_written_test(@registrant_3)
+
+      expect(@registrant_3.license_data).to eq ({written: false, license: false, renewed: false})
```

Comment:
Overall, great job testing for all the separate conditions that are going on here. Good test organization.